Exercícios/Para casa/entregas/Doria-fernandes/BankAccount.js

```diff
@@ -0,0 +1,122 @@
+const { Bank } = require('./Bank');
+const { Client } = require('./Client');
+
+
+class BankAccount {
+    client;
+    bank;
+    accountNumber;
+    agencyNumber;
+    #balance;
+    #qtdWithdrawal;
+    #withdrawalTax;
+
+
+    constructor(client, bank, accountNumber, agencyNumber) {
+        this.#balance = 0;
+        this.#qtdWithdrawal = 0;
+        this.#withdrawalTax = 7;
+        this.client = client;
+        this.bank = bank;
+        this.accountNumber = accountNumber;
+        this.agencyNumber = agencyNumber;
+    }
+
+    get balance() {
+        return this.#balance;
+    }
+
+    set balance(newBalance) {
+        return this.#balance = newBalance;
+    }
+
+    get qtdWithdrawal() {
+        return this.#qtdWithdrawal;
+    }
+
+    get withdrawalTax() {
+        return this.#withdrawalTax;
+    }
+
+    set withdrawalTax(newWithdrawaTax) {
+        this.#withdrawalTax += newWithdrawaTax;
+    }
+
+    credit(amount) {
+        this.#balance += amount;
+        console.log(`O saldo atualizado é de R$ ${this.#balance}`);
+    }
+
+    debit(amount) {
+        if (amount <= this.#balance) {
+            this.#balance -= amount;
+            console.log(`O saldo atualizado é de R$ ${this.#balance}`);
+        } else {
+            console.log("Não foi possível realizar esta operação")
+        }
+    }
+
+    transferTo(anotherAccount, amount) {
+        if (!(anotherAccount instanceof BankAccount)) {
+            console.log('Informe uma conta válida');
+            return;
+        } else if (amount > this.balance) {
```

Comment:
Eu sempre faço invertido, uso o this primeiro, acho que no exercício desta semana que teve um que fiz assim, usando atributo primeiro.  É interessante como há diversas formas e o mesmo resultado.